packages/components/src/internal/components/assay/actions.ts

```diff
@@ -100,21 +100,26 @@ export function uploadAssayRunFiles(data: IAssayUploadOptions): Promise<IAssayUp
         const batchFiles = collectFiles(batchProperties);
         const runFiles = collectFiles(runProperties);
 
+        const maxRowCount = Array.isArray(data.dataRows) ? data.dataRows.length : undefined;
+
         if (Utils.isEmptyObj(batchFiles) && Utils.isEmptyObj(runFiles)) {
             // No files in the data, so just go ahead and resolve so we run the import.
-            resolve(data);
+            resolve({...data, maxRowCount});
             return;
         }
 
         // If we're this far along we've got files so let's process them.
         let fileCounter = 0;
+        let maxFileSize = 0;
         const formData = new FormData();
         const fileNameMap = {}; // Maps the file name "fileN" to the run/batch property it belongs to.
 
         // TODO factor this out so the code isn't duplicate below for the runFiles case
         Object.keys(batchFiles).forEach(columnName => {
             const name = fileCounter === 0 ? 'file' : `file${fileCounter}`;
             const file = batchFiles[columnName];
+            if (file.size > maxFileSize)
+                maxFileSize = file.size;
```

Comment:
Isn't this effectively eliminating the maxFileSize?  Is that really what we want in all cases?

Reply:
This local field is used to determine the size of the import file, which is then returned to be used to (by comparing it with allowed size) determine if async mode should be used.

Reply:
Ah, ok.  Got it.